### PR TITLE
[READY] Adds firetrucks

### LINKED ISSUE #9031: Firetrucks
[Implementation]: A firetruck vehicle for atmos techs to spray reagents easily, could start loaded up with water. Maybe use minigun mechanics to keep spraying liquids. For balance purposes, make a reagent whitelist(water,nanofrost?)

[Why]: memes
could also give atmos tech more toys and love

[Resources]: No resource sadly

# Be sure to ask some maintainer to label this issue as a feature. Either ping them on our discord or PM them on our forum. Thank you for sharing your ideas!
@HippieStation/maintainers

### PR DESCRIPTION
**Pull request by Carbonhell**
https://imgur.com/a/i36xmgb

:cl:
add: Firetrucks. They're a vehicle with a water cannon loaded on them. You climb on it, drive to whatever fire you need to put down, get the controllers inhand and shoot firefighting foam at the fire till it dies. Normally it can be loaded with more foam or water, or anything if emagged. Admin-spawn only for now.
/:cl:

Closes #9031

### CODE REVIEW

**Comment by ghost** [comment]
+1

**Comment by bubberducky2** [comment]
me want firetruck go choo choo :+1: +1

**Comment by Carbonhell** [comment]
waiting for sprites

**Comment by Carbonhell** [comment]
Sprites added. It will be added to the map later on in a different PR,right now it's just inserted into the code

**Comment by Carbonhell** [comment]
Just needs a codecheck

File: hippiestation/code/modules/vehicles/firetruck.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		water_cannon.set_up(n = num_of_effects, loca = src, reagent_datum = reagents, the_target = target)
+		water_cannon.start()
+
+/obj/vehicle/ridden/firetruck/proc/spawn_water_cannon(mob/user)
+	if(!user.is_holding(controls)) // I didn't expect this to be so nicely readable
+		if(!user.put_in_hands(controls))
+			to_chat(user, "<span class='danger'>Your hands are full!</span>")
+			reset_controls_loc() // put_in_hands drops them on the floor on fail. This resets their location properly
+			return
+		to_chat(user, "<span class='notice'>The controls are in your hands now. Click or keep clicking on the turfs to shoot the cannon at them.")
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/vehicles/firetruck.dm:63]: This sounds a little confusing IMO. Might wanna just say "You activate the water cannon, click a turf to fire."

File: hippiestation/code/modules/vehicles/firetruck.dm [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		return
+	to_chat(user, "<span class='danger'>You already have the controls in your hands!</span>")
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/vehicles/firetruck.dm:65]: Same here "The controls" sounds a bit odd, try something like "You are already using the cannon"

File: hippiestation/code/modules/vehicles/firetruck.dm [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+	initialize_controller_action_type(/datum/action/vehicle/ridden/firetruck, VEHICLE_CONTROL_DRIVE)
+
+/obj/vehicle/ridden/firetruck/AllowClick()
+	return TRUE
+
+// Actions
+
+/datum/action/vehicle/ridden/firetruck
+	name = "Start Water Cannon"
+	desc = "Pick the water cannon controls"
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/vehicles/firetruck.dm:86]: Again use something like "Take control of the water cannon"

**Review by GuyonBroadway — APPROVED** [review]
Aside from some text grumbles this looks fine functionally speaking.

**Comment by Carbonhell** [comment]
@GuyonBroadway The action button spawns a special,abstract controls object in your hands though. That's what all the strings are referring to. Also,its icon is the same of a hand tele.

**Comment by Carbonhell** [comment]
so what needs to be changed exactly sorry

**Comment by Manly-Dwarf** [comment]
Add images for the sprites.

**Comment by Carbonhell** [comment]
can't you just use the dmi bo-
oh,it died,like usual,good fuckin riddance
firetruck_end

File: hippiestation/code/modules/vehicles/firetruck.dm (hunk 1 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/vehicles/firetruck.dm:63]: Ok so I think for maximum clarity this should be "You pick up the controls, click a tile to fire the water cannon."

File: hippiestation/code/modules/vehicles/firetruck.dm (hunk 2 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/vehicles/firetruck.dm:65]: Make this "You are already using the water cannon."

File: hippiestation/code/modules/vehicles/firetruck.dm (hunk 3 above)
Comment (GuyonBroadway) [reply hippiestation/code/modules/vehicles/firetruck.dm:86]: "Use the water cannon."

**Comment by Carbonhell** [comment]
Bump?

**Comment by biggiOS** [comment]
lil bump

**Comment by GuyonBroadway** [comment]
I'll take this as a "carbon thinks the text is fine" and let people gripe about grammar later.

**Pull request merged**